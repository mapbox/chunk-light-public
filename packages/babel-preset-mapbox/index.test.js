'use strict';

const babel = require('babel-core');

const preset = require('./index.js');

const transform = (code, options) =>
  babel.transform(code, {
    presets: [[require.resolve('./index.js'), options]],
    // This is needed 'babel-plugin-transform-react-jsx-source' throws an error without a filename
    filename: 'test'
  }).code;

test('Right order of configuration in production', () => {
  process.env.BABEL_ENV = 'production';
  expect(preset()).toMatchSnapshot();
  delete process.env.BABEL_ENV;
});

test('Right order of configuration in development', () => {
  process.env.BABEL_ENV = 'development';
  expect(preset()).toMatchSnapshot();
  delete process.env.BABEL_ENV;
});

test('Right order of configuration in test', () => {
  process.env.BABEL_ENV = 'test';
  expect(preset()).toMatchSnapshot();
  delete process.env.BABEL_ENV;
});

test('Passes opts correctly', () => {
  const config = preset(null, {
    'babel-preset-react': {
      abc: 'def'
    },
    'non-existent': {
      a: 'a'
    },
    'transform-class-properties': { xyz: '123' }
  });
  expect(config.presets).toContainEqual([
    'babel-preset-react',
    {
      abc: 'def'
    }
  ]);
  expect(config.plugins).toContainEqual([
    'babel-plugin-transform-class-properties',
    {
      xyz: '123'
    }
  ]);
  expect(config).toMatchSnapshot();
});

test('BROWSERSLIST env var works in non test env ', () => {
  process.env.BABEL_ENV = 'development';
  process.env.BROWSERSLIST = ['>0.25%', 'not ie 11', 'not op_mini all'].join(
    ','
  );
  const config = preset(null);
  expect(config.presets).toMatchSnapshot();
  delete process.env.BABEL_ENV;
  delete process.env.BROWSERSLIST;
});

test('Throws error when both BROWSERSLIST and babel-preset-env are provided ', () => {
  process.env.BROWSERSLIST = 'something';
  expect(() =>
    preset(null, {
      'babel-preset-env': {}
    })
  ).toThrowError();
  delete process.env.BROWSERSLIST;
});

test("BROWSERSLIST env var doesn't get injected in test env ", () => {
  process.env.BROWSERSLIST = ['>0.25%', 'not ie 11', 'not op_mini all'].join(
    ','
  );
  const config = preset(null);
  expect(config.presets).toMatchSnapshot();
  delete process.env.BROWSERSLIST;
});

test('Works with React', () => {
  expect(
    transform(`
        import React from 'react';
        const F = () => <Zoo/>;
    `)
  ).toMatchSnapshot();
});

test('Throws error when incorrect options are provided', () => {
  expect(() => transform(``, 45)).toThrowErrorMatchingSnapshot();
});

test('Applies options correctly to internal plugins and presets', () => {
  expect(
    transform(`const F = () => <Zoo/>;`, {
      'transform-react-jsx': {
        pragma: 'dom'
      }
    })
  ).toMatchSnapshot();
});
