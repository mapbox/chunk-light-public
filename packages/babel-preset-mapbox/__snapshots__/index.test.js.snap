// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applies options correctly to internal plugins and presets 1`] = `
"\\"use strict\\";

var _jsxFileName = \\"test\\";
const F = () => dom(Zoo, {
  __source: {
    fileName: _jsxFileName,
    lineNumber: 1
  },
  __self: undefined
});"
`;

exports[`BROWSERSLIST env var doesn't get injected in test env  1`] = `
Array [
  "babel-preset-react",
  Array [
    "babel-preset-env",
    Object {
      "targets": Object {
        "node": "current",
      },
    },
  ],
]
`;

exports[`BROWSERSLIST env var works in non test env  1`] = `
Array [
  "babel-preset-react",
  Array [
    "babel-preset-env",
    Object {
      "modules": false,
      "targets": Object {
        "browsers": ">0.25%,not ie 11,not op_mini all",
      },
      "useBuiltIns": true,
    },
  ],
]
`;

exports[`Passes opts correctly 1`] = `
Object {
  "plugins": Array [
    "babel-plugin-transform-es2015-destructuring",
    Array [
      "babel-plugin-transform-class-properties",
      Object {
        "xyz": "123",
      },
    ],
    Array [
      "babel-plugin-transform-object-rest-spread",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-react-jsx",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-runtime",
      Object {
        "helpers": false,
        "polyfill": false,
        "regenerator": true,
      },
    ],
    "babel-plugin-dynamic-import-node",
    "babel-plugin-transform-react-jsx-source",
    "babel-plugin-transform-react-jsx-self",
  ],
  "presets": Array [
    Array [
      "babel-preset-react",
      Object {
        "abc": "def",
      },
    ],
    Array [
      "babel-preset-env",
      Object {
        "targets": Object {
          "node": "current",
        },
      },
    ],
  ],
}
`;

exports[`Right order of configuration in development 1`] = `
Object {
  "plugins": Array [
    "babel-plugin-transform-es2015-destructuring",
    "babel-plugin-transform-class-properties",
    Array [
      "babel-plugin-transform-object-rest-spread",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-react-jsx",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-runtime",
      Object {
        "helpers": false,
        "polyfill": false,
        "regenerator": true,
      },
    ],
    "babel-plugin-syntax-dynamic-import",
    "babel-plugin-transform-react-jsx-source",
    "babel-plugin-transform-react-jsx-self",
  ],
  "presets": Array [
    "babel-preset-react",
    Array [
      "babel-preset-env",
      Object {
        "modules": false,
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`Right order of configuration in production 1`] = `
Object {
  "plugins": Array [
    "babel-plugin-transform-es2015-destructuring",
    "babel-plugin-transform-class-properties",
    Array [
      "babel-plugin-transform-object-rest-spread",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-react-jsx",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-runtime",
      Object {
        "helpers": false,
        "polyfill": false,
        "regenerator": true,
      },
    ],
    "babel-plugin-syntax-dynamic-import",
    "babel-plugin-transform-react-remove-prop-types",
  ],
  "presets": Array [
    "babel-preset-react",
    Array [
      "babel-preset-env",
      Object {
        "modules": false,
        "useBuiltIns": true,
      },
    ],
  ],
}
`;

exports[`Right order of configuration in test 1`] = `
Object {
  "plugins": Array [
    "babel-plugin-transform-es2015-destructuring",
    "babel-plugin-transform-class-properties",
    Array [
      "babel-plugin-transform-object-rest-spread",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-react-jsx",
      Object {
        "useBuiltIns": true,
      },
    ],
    Array [
      "babel-plugin-transform-runtime",
      Object {
        "helpers": false,
        "polyfill": false,
        "regenerator": true,
      },
    ],
    "babel-plugin-dynamic-import-node",
    "babel-plugin-transform-react-jsx-source",
    "babel-plugin-transform-react-jsx-self",
  ],
  "presets": Array [
    "babel-preset-react",
    Array [
      "babel-preset-env",
      Object {
        "targets": Object {
          "node": "current",
        },
      },
    ],
  ],
}
`;

exports[`Throws error when incorrect options are provided 1`] = `"\`@mapbox/babel-preset-mapbox\` expects an object as an option. (While processing preset: \\"<PROJECT_ROOT>/packages/babel-preset-mapbox/index.js\\")"`;

exports[`Works with React 1`] = `
"'use strict';

var _jsxFileName = 'test';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const F = () => _react2.default.createElement(Zoo, {
        __source: {
                fileName: _jsxFileName,
                lineNumber: 3
        },
        __self: undefined
});"
`;
